pkg/server/pod_controller.go

```diff
@@ -368,29 +368,33 @@ func (c *PodController) loadRegistryCredentials(pod *api.Pod) (map[string]api.Re
 
 	// AWS is different, they require us to authenticate with IAM
 	// Do that auth and pass along the username and password
-	for i := 0; i < len(pod.Spec.Units); i++ {
-		server, _, err := util.ParseImageSpec(pod.Spec.Units[i].Image)
+	if err := api.ForAllUnitsWithError(pod, func(unit *api.Unit) error {
```

Comment:
we can consider in future giving this function a name and test it. Your call ;)

Reply:
Good idea, I pulled it up into a separate function and added a test for it :+1: